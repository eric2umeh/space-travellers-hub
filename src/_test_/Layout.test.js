// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Layout from '../components/layout/Layout';

describe('Test Layout component', () => {
  test('should match with snapshot', () => {
    const layout = renderer
      .create(
        <BrowserRouter>
          <Layout>
            <h2>Hello World</h2>
          </Layout>
        </BrowserRouter>,
      )
      .toJSON();

    expect(layout).toMatchSnapshot();
  });
  test('should match snapshot', () => {
    render(
      <BrowserRouter>
        <Layout>
          <h2>Hello World</h2>
        </Layout>
      </BrowserRouter>,
    );

    const layout = screen.getByText('Hello World');
    expect(layout).toBeInTheDocument();
  });
});
