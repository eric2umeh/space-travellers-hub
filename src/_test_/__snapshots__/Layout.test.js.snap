// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Layout component should match with snapshot 1`] = `
Array [
  <header
    className="header"
  >
    <nav
      className="nav"
    >
      <div
        className="brand"
      >
        <div
          className="logo_img"
        >
          <img
            alt="Space Travlers"
            src="logo.png"
          />
        </div>
        <div
          className="logo_txt"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            <h1>
              Space Traveler's Hub
            </h1>
          </a>
        </div>
      </div>
      <ul
        className="nav_items"
      >
        <li>
          <a
            aria-current="page"
            className="active"
            href="/"
            onClick={[Function]}
            style={
              Object {
                "textDecoration": "underline",
              }
            }
          >
            Rockets
          </a>
        </li>
        <li>
          <a
            className=""
            href="/missions"
            onClick={[Function]}
          >
            Missions
          </a>
        </li>
        <li>
          <a
            className=""
            href="/profile"
            onClick={[Function]}
          >
            My Profile
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <main
    className="main"
  >
    <h2>
      Hello World
    </h2>
  </main>,
]
`;
